src/PhraseApp.php

```diff
@@ -0,0 +1,251 @@
+            }
+
+            $this->client->request('translation.update', [
+                'project_id' => $this->projectId,
+                'id' => $id,
+                'content' => $message->getDomain() . '::' . $message->getTranslation(),
+            ]);
+        } catch (\Throwable $e) {
+            throw new StorageException($e->getMessage());
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function update(Message $message)
+    {
+        $this->create($message);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function delete($locale, $domain, $key)
+    {
+        try {
+            $response = $this->client->request('key.search', [
+                'project_id' => $this->projectId,
+                'locale_id' => $this->getLocaleId($locale),
+                'q' => $domain . '::' . $key,
+            ]);
+
+            foreach ($response as $keyName) {
+                if ($keyName['name'] === $key) {
+                    $keyId = $key['id'];
+                    break;
+                }
+            }
+
+            if (! isset($keyId)) {
+                return;
+            }
+
+            $this->client->request('key.destroy', [
+                'project_id' => $this->projectId,
+                'id' => $keyId
+
+            ]);
+        } catch (\Throwable $e) {
+            throw new StorageException($e->getMessage());
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function export(MessageCatalogueInterface $catalogue)
+    {
+        $locale = $catalogue->getLocale();
+
+        foreach ($catalogue->getDomains() as $domain) {
+            try {
+                $response = $this->client->request('locale.download', [
+                    'project_id' => $this->projectId,
+                    'id' => $this->getLocaleId($locale),
+                    'tag' => $domain,
+                    'file_format' => 'symfony_xliff'
+                ]);
+            } catch (\Throwable $e) {
+                throw new StorageException($e->getMessage());
+            }
+
+            $data = $response['text'];
+            /* @var \GuzzleHttp\Stream\Stream $data */
+            $catalogue->addCatalogue(XliffConverter::contentToCatalogue($data->getContents(), $locale, $domain));
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function import(MessageCatalogueInterface $catalogue)
+    {
+        foreach ($this->localeToIdMapping as $locale => $localeId) {
+            foreach ($catalogue->getDomains() as $domain) {
+                $data = XliffConverter::catalogueToContent($catalogue, $domain);
+                $file = sys_get_temp_dir() . '/' . $domain . '_' . $locale . '.xlf';
+                file_put_contents($file, $data);
+
+                try {
+                    /* I could not get guzzle to work with this, so fallback to curl for now
+                     *
+                    $response = $this->client->request('upload.create', [
```

Comment:
Im not sure. This seams to be an issue with the client. Maybe @nediam knows why?